### PR TITLE
Added baremetal support

### PR DESCRIPTION
**Pull request by aajdinov**

### CODE REVIEW

**Comment by aajdinov** [comment]
@alibazlamit Could you please review these changes?

File: OneAndOne.Client/Endpoints/Servers/Servers.cs [hunk 1]
```diff
@@ -70,6 +70,55 @@ public List<ServerResponse> Get(int? page = null, int? perPage = null, string so
...
 
         /// <summary>
+        /// Returns a list of your baremetal servers.
+        /// </summary>
+        /// <param name="page">Allows to use pagination. Sets the number of servers that will be shown in each page.</param>
+        /// <param name="perPage">Current page to show.</param>
+        /// <param name="sort">Allows to sort the result by priority:sort=name retrieves a list of elements ordered by their names.sort=-creation_date retrieves a list of elements ordered according to their creation date in descending order of priority.</param>
+        /// <param name="query">Allows to search one string in the response and return the elements that contain it. In order to specify the string use parameter q:    q=My server</param>
+        /// <param name="fields">Returns only the parameters requested: fields=id,name,description,hardware.ram</param>
+        public List<ServerResponse> GetBaremetal(int? page = null, int? perPage = null, string sort = null, string query = null, string fields = null)
```
Comment (alibazlamit) [inline OneAndOne.Client/Endpoints/Servers/Servers.cs]: This return a list of bare metal models not servers

File: OneAndOne.Client/Endpoints/Servers/Servers.cs [hunk 2]
```diff
@@ -180,6 +229,31 @@ public ServerResponse Show(string server_id)
         }
 
         /// <summary>
+        /// Returns baremetal server's information.
+        /// </summary>
+        /// <param name="server_id">Unique server's identifier.</param>
+        /// 
+        public ServerResponse ShowBaremetal(string server_id)
```
Comment (alibazlamit) [inline OneAndOne.Client/Endpoints/Servers/Servers.cs]: same comment as above

**Comment by aajdinov** [comment]
Per the requirements file:

> - GET /servers, GET /servers/{id} and all /servers 20x response:
>   --> New parameter "server_type" with possible values "cloud" or "baremetal"
>   --> New parameter "baremetal_model_id" within hardware with the ID of the baremetal model
>   --> New parameter "ipv6_range" with the IPv6 range assigned to the server
>   --> New parameter "hostname" with the hostname assigned to the server

Since there are parameters `server_type` (which include `baremetal` as server type) and `baremetal_model_id` within `Hardware` as a part of the `ServerResponse`, are we sure that a new model is required? `ServerResponse` model does have the same fields (and additional ones).

**Comment by edevenport** [comment]
@aajdinov Can we resolve the merge conflicts?

**Comment by aajdinov** [comment]
Opening PR from SPC master.

**Pull request closed without merging**